### ISSUE TITLE
Markdown rendering not working in Adaptive Cards in local build

### ISSUE THREAD

**Issue by arashberlin** [Microsoft/BotFramework-WebChat#972]
Hi there,

I am working on a customized version of the Web-Chat [Title, color & logo].

Adaptive Cards with Markdown Url do not render correct, when I use the code and build myself and use the botchat.js

This how do I build [as describen in the [readme](https://github.com/Microsoft/BotFramework-WebChat#building-web-chat)]:

I downloaded the 0.13.1 version 
npm install
npm run build

and copy the botchat.js  to my destition folder 

When I use the  [botchat.js CDN ](https://cdn.botframework.com/botframework-webchat/latest/botchat.js) the Markdown render correct

Here my HTML Setup:

![image](https://user-images.githubusercontent.com/12381397/39699616-4334cbcc-51fa-11e8-9142-a8c707fafa3f.png)

![image](https://user-images.githubusercontent.com/12381397/39699627-4784396a-51fa-11e8-84f0-b71c62496c42.png)

Here the result:

with local botchat.js 

![image](https://user-images.githubusercontent.com/12381397/39699646-64bfee7a-51fa-11e8-9226-c6b528f87c53.png)

 [botchat.js CDN ](https://cdn.botframework.com/botframework-webchat/latest/botchat.js)

![image](https://user-images.githubusercontent.com/12381397/39699657-6e7a8f10-51fa-11e8-8b90-e95e0d0c2ea2.png)

Any idea how can I fix the issue?  Is this a bug?

Many Thanks

**Comment by Alaiven** [comment]
Also relates to #949

**Comment by danmarshall** [comment]
Can you provide the markdown that is causing the bug?

**Comment by arashberlin** [comment]
Hi @danmarshall thank you for the response. Many thanks to help me.

Is this a bug in **Adapative Card?**, because I get the issue now in the emulator, or has my code an issue?
 :(


Markdown for Url is like this: 
![image](https://user-images.githubusercontent.com/12381397/40103011-5fb1d6e8-58ec-11e8-8037-ff271566ed86.png)



This is the methode how I create the Adapative Card:
`
public static Attachment GetAdapativeCard(string Title, string Abstract, List<AdaptiveAction> Actions, string Subtitle, string Url = null, string ImageUrl = null)
        {

            AdaptiveCard card = new AdaptiveCard();

            // Specify speech for the card.  
            card.Speak = $"<s>{Title}</s>";

            if (Actions != null)
            {
                card.Actions = Actions;
            }


            if (ImageUrl != null)
            {
                card.Body.Add(new AdaptiveImage()
                {
                    Url = new Uri(ImageUrl),
                    Size = AdaptiveImageSize.Stretch,
                    Style = AdaptiveImageStyle.Default,
                    AltText = "Image",
                    HorizontalAlignment = AdaptiveHorizontalAlignment.Left,
                    SelectAction = new AdaptiveOpenUrlAction()
                    {
                        Url = Url != null ? new Uri(Url) : new Uri(ImageUrl)
                    }

                });
            }


            card.Body.Add(new AdaptiveTextBlock()
            {
                Text = Url != null ? $"[{Title}]({Url})" : Title,
                Size = AdaptiveTextSize.Large,
                Weight = AdaptiveTextWeight.Bolder
            });

            if (Subtitle != null)
            {
                card.Body.Add(new AdaptiveTextBlock()
                {
                    Text = Subtitle,
                    IsSubtle = true,
                    Separator = true
                });
            }


            card.Body.Add(new AdaptiveTextBlock()
            {
                Text = Abstract
            });



            // Create the attachment with adapative card.  
            Attachment attachment = new Attachment()
            {
                ContentType = AdaptiveCard.ContentType,
                Content = card
            };
            return attachment;
        }
`

![image](https://user-images.githubusercontent.com/12381397/40102823-cbea292e-58eb-11e8-81ca-a861892c65ae.png)

![image](https://user-images.githubusercontent.com/12381397/40102940-1b767a7e-58ec-11e8-98c6-0e91dfd3ea0b.png)

**Comment by compulim** [comment]
Note to fixer: in order to use Markdown in Adaptive Cards, user need to load markdown-it.js.

**Comment by amirt-microsoft** [comment]
Is there any updates on this issue @danmarshall ?

The solution suggested by @arashberlin is not applicable for me because I have customer's html pointing to my build of botchat.js, that should support markdown.

I see that when I'm using:
https://cdn.botframework.com/botframework-webchat/latest/botchat.js
markdown rendering is working fine for cards, while my own build from the latest master branch doesn't.
Can you explain why it's different? 

Thanks!

**Comment by compulim** [comment]
@amirt-microsoft This is interesting. The build we publish to NPM and CDN is exactly the same as on GitHub. We build it on Travis CI so everything is very transparent.

**Comment by compulim** [comment]
@amirt-microsoft This is interesting. The build we publish to NPM and CDN use the source code from GitHub and built on Travis CI.

Can you attach a minimal repro of your Adaptive Cards JSON so I can repro it?

**Comment by compulim** [comment]
@amirt-microsoft Is it dupe of #949?

**Comment by amirt-microsoft** [comment]
1. @compulim , yes, this is indeed a duplicate of  #949 

2. @compulim attaching a json for you to try:
[adaptiveCard.txt](https://github.com/Microsoft/BotFramework-WebChat/files/2140426/adaptiveCard.txt)

3. @compulim , can you share your build process definition for me to compare?

**Comment by compulim** [comment]
For build process, look at [.travis.yml](https://github.com/Microsoft/BotFramework-WebChat/blob/master/.travis.yml). It is as simple as:

> Make sure you are on Node 9, we will bump to Node 10 when we bump Sass.

```sh
npm install
npm run install-test
npm run build
npm run build-test
```

You can look at the latest build log on [Travis CI](https://travis-ci.org/Microsoft/BotFramework-WebChat). Although it build on Linux, I always build on Windows.

**Comment by amirt-microsoft** [comment]
Hi @compulim ,
The issue has escalated. 
now after you have released 0.14, and updated the build, it's also not rendering markdown in cards using the link below:
https://cdn.botframework.com/botframework-webchat/latest/botchat.js

so now both local and using your latest build is not working.

This is a blocker for our team, Is anyone looking into this issue?

Thanks

**Comment by compulim** [comment]
Will fix in PR #1034. Since we bundled Markdown-It in Web Chat already, we will expose it to `window.markdownit` so Adaptive Cards can use it.

**Comment by amirt-microsoft** [comment]
Thanks @compulim !
looking forward to get a new release with the fix

**Comment by amirt-microsoft** [comment]
Hi @compulim, the pull request has been open for 3 weeks now. are you going to complete it anytime soon?
are you going to issue a new release anytime soon?
Thanks!

**Comment by compulim** [comment]
@amirt-microsoft Sorry can't make it sooner, waiting for people to review the PR. I just pinged and added a few, hope it will speed up the PR. 😓

**Comment by amirt-microsoft** [comment]
Thanks @compulim

**Comment by compulim** [comment]
Web Chat [0.14.1](https://www.npmjs.com/package/botframework-webchat/v/0.14.1) is out with the fix #1034.
